applications/trilinos_application/python_scripts/trilinos_import_model_part_utility.py

```diff
@@ -59,14 +74,25 @@ def ExecutePartitioningAndReading(self):
             ## Read the new generated *.mdpa files
             KratosMultiphysics.ModelPartIO(mpi_input_filename).ReadModelPart(self.main_model_part)
 
+        elif input_type == "rest":
+            from trilinos_restart_utility import TrilinosRestartUtility as RestartUtility
+            restart_settings = self.settings.Clone()
+
+            restart_settings.RemoveValue("input_type")
+
+            restart_settings.AddEmptyValue("set_mpi_communicator")
+            restart_settings["set_mpi_communicator"].SetBool(False)
+
+            if not restart_settings.Has("restart_load_file_label"):
+                raise Exception('"restart_load_file_label" must be specified when starting from a restart-file!')
+
+            RestartUtility(self.main_model_part, restart_settings).LoadRestart()
+
         else:
             raise Exception("Other input options are not yet implemented.")
 
     def CreateCommunicators(self):
-        ## Construct and execute the MPICommunicator
-        KratosMetis.SetMPICommunicatorProcess(self.main_model_part).Execute()
```

Comment:
Are you sure that this can be removed?

Reply:
The metis process only (and only) replaces the `Communicator` for the `MPICommunicator`
=> [see here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/metis_application/custom_processes/set_mpi_communicator_process.h#L119)

this is also the first thing the `ParallelFillCommunicator` does, see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/trilinos_application/custom_utilities/parallel_fill_communicator.h#L324-L325)
The `ParallelFillCommunicator` also computes the graphs for the SubModelParts according to @jcotela

Reply:
Perfect! Thanks for the explanation  @philbucher